sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -217,7 +219,7 @@ trait FileSourceScanLike extends DataSourceScanExec {
         vectorTypes ++
           // for column-based file format, append metadata column's vector type classes if any
           metadataColumns.map { metadataCol =>
-            if (FileFormat.isConstantMetadataAttr(metadataCol.name)) {
+            if (FileFormat.isConstantMetadataAttr(metadataCol)) {
```

Comment:
do we still need it? `lazy val metadataColumns` is guaranteed to be only constant metadata columns.

Reply:
That is an excellent point,  updated this and also renamed the variable to make it clear that we only handle constant metadata here